tests/virt_autotest/host_upgrade_step2_run.pm

```diff
@@ -0,0 +1,46 @@
+# SUSE's openQA tests
+#
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+use base "host_upgrade_base";
+#use virt_utils qw(set_serialdev);
+use testapi;
+use strict;
+
+sub get_script_run() {
+    my $self = shift;
+
+    my $pre_test_cmd = $self->get_test_name_prefix;
+    $pre_test_cmd .= "-run 02";
+
+    return "$pre_test_cmd";
+}
+
+sub run() {
+    my $self = shift;
+
+    # Set the correct serial dev for ipmi xen and non-xen host according to the installed product release
+    # &virt_utils::set_serialdev();
+
+    # Got script run according to different kind of system
+    my $pre_test_cmd = $self->get_script_run();
+
+    # Execute script run
+    my $ret = $self->execute_script_run($pre_test_cmd, 36000);
```

Comment:
Yes, MAX_JOB_TIME would be possible to set in either workers or test_suites but the problem I see is what we have already seen with the "qa_acceptance" performance tests: Whole builds would not finish because some tests are running for 9 hours and get retriggered on failure so multiple days of waiting. In the end, it was errors in the tests that made it fail way before 9 hours but it wasn't noticed because the main "wait_serial" call would only timeout after 9h. So I did not follow the full tests implementation but I see two possibilities to improve, maybe there are more:

1. If the timeout is so high because you are relying on slow network connection between bejing and nuremberg or provo, set a much lower value and set TIMEOUT_SCALE in your local environment, e.g. on the corresponding machine definitions.
2. Better separate your tests into individual scenarios that can go after each other using the "START_AFTER" setting and relying on generated assets, e.g. one job is only creating an installation on a HDD image and uploads the HDD image as "asset". Then another job can take this HDD image, download it, and continue from. Examples are [gnome_create_image](https://openqa.suse.de/tests/449704) generating the HDD image and [sles+extra_tests](https://openqa.suse.de/tests/449840) using the HDD image.